rasa/shared/nlu/training_data/loading.py

```diff
@@ -60,7 +60,10 @@ def load_data(resource_name: Text, language: Optional[Text] = "en") -> "Training
     elif len(data_sets) == 1:
         training_data = data_sets[0]
     else:
-        training_data = data_sets[0].merge(*data_sets[1:])
+        first_training_data_instance = data_sets[0]
+        if first_training_data_instance is None:
```

Comment:
I think the proper way to fix it is either to `cast(List[TrainingData], data_sets)` at line 57 or rename the `data_sets` variable (because its type changed and mypy can't pick it up)